src/extension/components/button-transaction.tsx

```diff
@@ -0,0 +1,188 @@
+import React, { useEffect, useState } from 'react';
+import type { TxDetails } from '../../domain/transaction';
+import { TxType } from '../../domain/transaction';
+import { formatDecimalAmount, fromSatoshi, fromSatoshiStr } from '../utility';
+import TxIcon from './txIcon';
+import moment from 'moment';
+import Modal from './modal';
+import { Transaction } from 'liquidjs-lib';
+import type { Asset } from '../../domain/asset';
+import type { BlockHeader } from '../../background/utils';
+import AssetIcon from './assetIcon';
+import { confidentialValueToSatoshi } from 'liquidjs-lib/src/confidential';
+import Button from './button';
+import Browser from 'webextension-polyfill';
+import {
+  appRepository,
+  useSelectNetwork,
+  walletRepository,
+} from '../../infrastructure/storage/common';
+import { makeURLwithBlinders } from '../../utils';
+
+function txTypeFromTransfer(transfer?: number): TxType {
+  if (transfer === undefined) return TxType.Unknow;
+  if (transfer === 0) return TxType.SelfTransfer;
+  if (transfer > 0) return TxType.Deposit;
+  return TxType.Withdraw;
+}
+
+interface Props {
+  txDetails?: TxDetails;
+  assetSelected: Asset;
+}
+
+const ButtonTransaction: React.FC<Props> = ({ txDetails, assetSelected }) => {
+  const [modalOpen, setModalOpen] = useState(false);
+  const [transfer, setTransfer] = useState<{ amount: number; type: TxType }>();
+  const [feeAmount, setFeeAmount] = useState<number>();
+  const [txID, setTxID] = useState<string>();
+  const [blockHeader, setBlockHeader] = useState<BlockHeader>();
+  const [isLoading, setIsLoading] = useState(true);
+  const network = useSelectNetwork();
+
+  useEffect(() => {
+    // compute transfer
+    (async () => {
+      if (!txDetails?.hex) return;
+      const transaction = Transaction.fromHex(txDetails.hex);
+      const txID = transaction.getId();
+      setTxID(txID);
+
+      let transferAmount = 0;
+
+      for (let outIndex = 0; outIndex < transaction.outs.length; outIndex++) {
+        const output = transaction.outs[outIndex];
+        if (output.script.length === 0) {
+          setFeeAmount(confidentialValueToSatoshi(output.value));
+          continue;
+        }
+
+        const data = await walletRepository.getOutputBlindingData(txID, outIndex);
+        if (!data || !data.blindingData) continue;
+        if (data.blindingData.asset === assetSelected.assetHash) {
+          transferAmount += data.blindingData.value;
+        }
+      }
+
+      for (let inIndex = 0; inIndex < transaction.ins.length; inIndex++) {
+        const input = transaction.ins[inIndex];
+        const data = await walletRepository.getOutputBlindingData(
+          Buffer.from(input.hash).reverse().toString('hex'),
```

Comment:
`input.hash` is already a Buffer. Any reason to use Buffer.from() ?

Reply:
it prevents mutation on `input.hash` (reverse does not copy the buffer)

Reply:
Ok thanks